pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +183,13 @@ type HTTPSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// KubernetesSDConfig allow retrieving scrape targets from Kubernetes' REST API
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config
+// +k8s:openapi-gen=true
+type KubernetesSDConfig struct {
+	// Role is role of the service in Kubernetes
+	// Currently only Node role is supported
+	//+kubebuilder:validation:Enum=Node
+	Role string `json:"role,omitempty"`
```

Comment:
Do we plan to support other kinds of roles like pod|service?

Reply:
I believe that would be future so that we don't replicate fields in all monitors/probe for new prometheus added features. 
May be once ScrapeConfig is v1 we can add new features only to ScrapeConfig CRD or we need to come up with migration plan to move monitors to ScrapeConfig. But it needs further discussions.

But thats my understanding from proposal https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202212-scrape-config.md :)